desc/compute/_stability.py

```diff
@@ -231,3 +231,513 @@ def _magnetic_well(params, transforms, profiles, data, **kwargs):
         0,  # coefficient of limit is V_r / V_rr, rest is finite
     )
     return data
+
+
+@register_compute_fun(
+    name="ideal_ball_gamma1",
+    label="\\gamma^2",
+    units=" ",
+    units_long=" ",
+    description="ideal ballooning growth rate" + "requires data along a field line",
+    dim=1,
+    params=["Psi"],
+    transforms={"grid": []},
+    profiles=[],
+    coordinates="rtz",
+    data=[
+        "a",
+        "g^aa",
+        "g^ra",
+        "g^rr",
+        "cvdrift",
+        "cvdrift0",
+        "|B|",
+        "B^zeta",
+        "p_r",
+        "phi",
+        "iota",
+        "iota_r",
+        "psi",
+        "psi_r",
+        "rho",
+    ],
+)
+def _ideal_ballooning_gamma1(params, transforms, profiles, data, *kwargs):
+    """
+    Ideal-ballooning growth rate finder.
+
+    This function uses a finite-difference method
+    to calculate the maximum growth rate against the
+    infinite-n ideal ballooning mode. The equation being solved is
+
+    d/dζ(g dX/dζ) + c X = lam f X, g, f > 0,
+
+    where
+
+    𝛋 = 𝐛 ⋅∇ 𝐛
+    g = a_N^3 * B_N * (b ⋅∇ζ) * |∇α|², / B,
+    c = a_N/B_N * (1/ b ⋅∇ζ) * dψ/dρ * dp/dψ * (b × 𝛋) ⋅|∇α|/ B**2,
+    f = a_N * B_N^3 *|∇α|² / B^3 * (1/ b ⋅∇ζ) ,
+
+    are needed along a field line to solve the ballooning equation once.
+
+    To obtain the parameters g, c, and f, we need a set of parameters
+    provided in the list ``data`` above. Here's a description of
+    these parameters:
+
+    - a: minor radius of the device
+    - g^aa: |grad alpha|^2, field line bending term
+    - g^ra: (grad alpha dot grad rho) integrated local shear
+    - g^rr: |grad rho|^2 flux expansion term
+    - cvdrift: geometric factor of the curvature drift
+    - cvdrift0: geoetric factor of curvature drift 2
+    - |B|: magnitude of the magnetic field
+    - B^zeta: inverse of the jacobian
+    - p_r: dp/drho, pressure gradient
+    - phi: coordinate describing the position in the toroidal angle
+    along a field line
+    """
+    rho = data["rho"]
+
+    psi_b = params["Psi"] / (2 * jnp.pi)
+    a_N = data["a"]
+    B_N = 2 * psi_b / a_N**2
+
+    N_zeta0 = int(11)
```

Comment:
this should be a user specify-able parameter, maybe as part of `kwargs`

Reply:
How? Is there an example somewhere?

Reply:
in theory it could just be `N_zeta0 = kwargs.get("N_zeta0", 11)` but I think in this case N_zeta0 is really determined by the grid right? So does this work if a different grid is passed in?

Reply:
Which grid? N_zeta0 is independent of N_zeta.